addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1286,7 +1288,7 @@ const Wysiwyg = Widget.extend({
                 }
             }
             this.linkTools.destroy();
-            if (anchorNode) {
+            if (anchorNode && !(ev && this.$editable[0].contains(ev.target))) {
```

Comment:
Rather than this why no make sure `anchorNode` is not set to `link` for such a case when `editable` contains `selection.anchorNode`?
Instead of
```js
[anchorNode, focusNode] = commonBlock && link.contains(commonBlock) ? [commonBlock, commonBlock] : [link, link];
```
something like:
```js
  if (commonBlock && link.contains(commonBlock)) {
      [anchorNode, focusNode] = [commonBlock, commonBlock];
  } else if (!this.$editable[0].contains(selection.anchorNode)) {
      [anchorNode, focusNode] = [link, link];
  }
```

Reply:
@adsa-odoo Is there a particular reason why this comment was left unaddressed? I like @deso-odoo's suggestion if it can get rid of the event parameter givent to `destroyLinkTools`.

Reply:
@dmo-odoo Yes, when I try @deso-odoo's suggestion then it works but in [this](https://github.com/odoo/odoo/blob/16.0/addons/website/static/tests/tours/link_tools.js#L35C5-L49C7) tour link removal step fails.

**Why the link is not getting removed in tour by [DESO]'s fix?**

Here in first step, after link creation image is clicked using `clickOnImgStep`. At this time selection is on image's div container. After that newly created link is clicked and popover is opened. At that time selection is still on image's div container and not on link. When trying to remove link using popover, it doesn't get removed because there is no selection on link.
Without [DESO]'s fix it works because when image is clicked the newly created link will be selected because of `destroyLinkTools`.
However, in my fix tour passes because when `.fa-chain-broken` icon is clicked the selection is still not in the link but `destroyLinkTools` gets called first and it will select the link (as `.fa-chain-broken` icon is not the part of editor and link will be selected) and link gets removed.

So we either keep @deso-odoo's fix or my fix. If we keep @deso-odoo's fix then we probably need to change the tour. For ex. simulate selection on link before removing the link.

Reply:
@adsa-odoo Well, the one particular thing I don't want to do is passing an event to `destroyLinkTools`. If one of the two alternatives can work without passing an event to `destroyLinkTools` then I choose that alternative.

Reply:
@dmo-odoo I've updated the code, thanks!